code/game/gamemodes/cult/talisman.dm

```diff
@@ -121,112 +99,113 @@ Rite of Disorientation
-
-/obj/item/weapon/paper/talisman/teleport/invoke(mob/living/user)
-	var/list/possible_runes = list()
-	for(var/obj/effect/rune/teleport/R in teleport_runes)
-		if(R.keyword == src.keyword)
-			possible_runes.Add(R)
-	if(!possible_runes.len)
-		user << "<span class='cultitalic'>There are no Teleport runes with the same keyword!</span>"
-		log_game("Teleportation talisman failed - no teleport runes of the same keyword")
-		uses++ //To prevent deletion
-		return
-	var/chosen_rune = pick(possible_runes)
+
+/obj/item/weapon/paper/talisman/teleport/invoke(mob/living/user, successfuluse = 1)
+	var/list/potential_runes = list()
+	var/list/teleportnames = list()
+	var/list/duplicaterunecount = list()
+	for(var/R in teleport_runes)
+		var/obj/effect/rune/teleport/T = R
+		var/resultkey = T.listkey
+		if(resultkey in teleportnames)
+			duplicaterunecount[resultkey]++
+			resultkey = "[resultkey] ([duplicaterunecount[resultkey]])"
+		else
+			teleportnames.Add(resultkey)
+			duplicaterunecount[resultkey] = 1
+		potential_runes[resultkey] = T
+
+	if(!potential_runes.len)
+		user << "<span class='warning'>There are no valid runes to teleport to!</span>"
+		log_game("Teleport talisman failed - no other teleport runes")
+		return ..(user, 0)
+
+	if(user.z > ZLEVEL_SPACEMAX)
+		user << "<span class='cultitalic'>You are not in the right dimension!</span>"
+		log_game("Teleport talisman failed - user in away mission")
+		return ..(user, 0)
+
+	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
+	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
+	if(!actual_selected_rune)
+		return ..(user, 0)
 	user.visible_message("<span class='warning'>Dust flows from [user]'s hand, and they disappear in a flash of red light!</span>", \
 						 "<span class='cultitalic'>You speak the words of the talisman and find yourself somewhere else!</span>")
-	if(user.buckled)
-		user.buckled.unbuckle_mob(user,force=1)
-	user.loc = get_turf(chosen_rune)
-
-/obj/item/weapon/paper/talisman/teleport/New()
-	..()
-	spawn(1) //To give the keyword time to change from the imbue rune
-		info += keyword
+	user.forceMove(get_turf(actual_selected_rune))
+	return ..()
 
-/obj/item/weapon/paper/talisman/teleport/examine(mob/user)
-	..()
-	if(iscultist(user) && keyword)
-		user << "<b>Keyword:</b> [keyword]"
 
-//Rite of Knowledge: Same as rune, but has two uses
 /obj/item/weapon/paper/talisman/summon_tome
 	cultist_name = "Talisman of Tome Summoning"
 	cultist_desc = "A one-use talisman that will call an untranslated tome from the archives of the Geometer."
+	color = "#1d0e0e" // red-black
 	invocation = "N'ath reth sh'yro eth d'raggathnor!"
 	health_cost = 1
 
-/obj/item/weapon/paper/talisman/summon_tome/invoke(mob/living/user)
+/obj/item/weapon/paper/talisman/summon_tome/invoke(mob/living/user, successfuluse = 1)
```

Comment:
This proc needs a `. = ..()` in-between lines 152-153 so that it properly invokes.